src/runtime/threading_backend.cc

```diff
@@ -321,12 +350,25 @@ class ThreadGroup::Impl {
     }
   }
 
+#ifndef __hexagon__
+  pid_t Tid() {
+#if defined(_WIN32)
+    return GetCurrentThreadId();
+#else
+    return syscall(SYS_gettid);
+#endif
+  }
+  void SetTid(size_t index) { threads_tid_[index] = Tid(); }
+  pid_t GetTid(size_t thread_index) { return threads_tid_[thread_index]; }
```

Comment:
This not fix the timing issue, that means SetTid may happen after GetTid, and threads_tid_[thread_index] will point to a random value or a '0' value depend on the compiler settings.

Reply:
If getid() is called before settid(). Gettid will return 0. Call sched_setaffinity will return -1.

Reply:
Returns -1, indicating that the affinity setting failed. In line with expectations

Reply:
> Returns -1, indicating that the affinity setting failed. In line with expectations

 this sounds not make sense, why "affinity setting failed" is " In line with expectations", why need to expect "affinity setting failed"?

Reply:
SetTid after Gettid means we set affinity for a nonexistent thread. I think should return -1

Reply:
"SetTid after Gettid" not means "we set affinity for a nonexistent thread", thread already there it just not yet execute the "SetTid" function.